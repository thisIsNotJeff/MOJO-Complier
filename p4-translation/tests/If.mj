{
    var x := 0;

    if x == 1 {
        x := 10;
    } else if x < 4 {
        x := 20;
    } else {
        x := 30;
    }
}
