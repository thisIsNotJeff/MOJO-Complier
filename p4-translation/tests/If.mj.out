	.text
If:
MOVE(
 TEMP If.1.x,
 CONST 0),
BEQ(
 TEMP If.1.x,
 CONST 1,
 L.1, L.2),
LABEL L.1,
MOVE(
 TEMP If.1.x,
 CONST 10),
JUMP(
 NAME L.3),
LABEL L.2,
BLT(
 TEMP If.1.x,
 CONST 4,
 L.4, L.5),
LABEL L.4,
MOVE(
 TEMP If.1.x,
 CONST 20),
JUMP(
 NAME L.6),
LABEL L.5,
MOVE(
 TEMP If.1.x,
 CONST 30),
LABEL L.6,
LABEL L.3


